src/url-replacements.js

```diff
@@ -111,15 +119,42 @@ class UrlReplacements {
    */
   expand(url, opt_data) {
     const expr = this.getExpr_();
-    return url.replace(expr, (match, name) => {
-      let val = this.replacements_[name](opt_data);
+    let replacementPromise;
+    const encodeValue = val => {
       // Value 0 is specialcased because the numeric 0 is a valid substitution
       // value.
       if (!val && val !== 0) {
         val = '';
       }
       return encodeURIComponent(val);
+    };
+    url = url.replace(expr, (match, name, arg) => {
+      const val = this.replacements_[name](opt_data, arg);
+      // In case the produced value is a promise, we don't actually
+      // replace anything here, but do it again when the promise resolves.
+      if (val instanceof Promise) {
```

Comment:
That could in theory run into the same problem we had with Promise polyfill and native like we had with fetch. I just recently cleaned those up. Should we err here as well?